src/game/server/entities/character.cpp

```diff
@@ -2580,6 +2647,7 @@ bool CCharacter::TakeDamage(vec2 Force, int Dmg, int From, int Weapon, int Mode)
 {
 /* INFECTION MODIFICATION START ***************************************/
 
+	//KillerPlayer
 	CPlayer* pKillerPlayer = GameServer()->m_apPlayers[From];
```

Comment:
This is great that you have found this variable and reuse it. It would also be great to check that this pointer is not null before we continue. Something like
```
if (!pKillerPlayer || !pKillerPlayer->GetCharacter()) {
	return false;  // something is wrong with the killer player
}
```
and remove other checks below.

Reply:
Unfortunately, it was a very bad idea. It causes boomer not to cause any damage because of `GetCharacter()` check Let me try and fix that.

Reply:
I dident see your comment, i already wrote a fix for it